From 5adc295e3bb7ad246de181cb0711b8f81d53d399 Mon Sep 17 00:00:00 2001
From: Daryl Denson <daryldenson0405@gmail.com>
Date: Wed, 21 Oct 2026 09:30:54 +0200
Subject: [PATCH 2/2] husky: point DerpFest's Updater at our OTA repo

DerpFest's Updater reads DerpFest-AOSP/Updater-Stuff/<device>.json,
which for husky lists the official builds. They're signed with other
keys, so this unofficial build would be offered updates it can't
install. A static RRO points it at denson9874/evolution_ota
(derpfest/builds, derpfest/changelogs) instead, as kodiak does.
---
 lineage_husky.mk                                      |  4 ++++
 rro_overlays/UpdaterOverlayDerpHusky/Android.bp       |  9 +++++++++
 .../UpdaterOverlayDerpHusky/AndroidManifest.xml       | 11 +++++++++++
 .../UpdaterOverlayDerpHusky/res/values/strings.xml    | 11 +++++++++++
 4 files changed, 35 insertions(+)
 create mode 100644 rro_overlays/UpdaterOverlayDerpHusky/Android.bp
 create mode 100644 rro_overlays/UpdaterOverlayDerpHusky/AndroidManifest.xml
 create mode 100644 rro_overlays/UpdaterOverlayDerpHusky/res/values/strings.xml

diff --git a/lineage_husky.mk b/lineage_husky.mk
index 475c271c..a8e531f2 100644
--- a/lineage_husky.mk
+++ b/lineage_husky.mk
@@ -41,3 +41,7 @@ PRODUCT_BUILD_PROP_OVERRIDES += \
     DeviceProduct=$(DEVICE_CODENAME)
 
 $(call inherit-product, $(VENDOR_PATH)/$(DEVICE_CODENAME)-vendor.mk)
+
+# DerpFest: unofficial build, so the Updater reads our OTA repo, not DerpFest's husky.json
+PRODUCT_PACKAGES += \
+    UpdaterOverlayDerpHusky
diff --git a/rro_overlays/UpdaterOverlayDerpHusky/Android.bp b/rro_overlays/UpdaterOverlayDerpHusky/Android.bp
new file mode 100644
index 00000000..5563eb61
--- /dev/null
+++ b/rro_overlays/UpdaterOverlayDerpHusky/Android.bp
@@ -0,0 +1,9 @@
+//
+// SPDX-FileCopyrightText: The LineageOS Project
+// SPDX-License-Identifier: Apache-2.0
+//
+
+runtime_resource_overlay {
+    name: "UpdaterOverlayDerpHusky",
+    product_specific: true,
+}
diff --git a/rro_overlays/UpdaterOverlayDerpHusky/AndroidManifest.xml b/rro_overlays/UpdaterOverlayDerpHusky/AndroidManifest.xml
new file mode 100644
index 00000000..5515172a
--- /dev/null
+++ b/rro_overlays/UpdaterOverlayDerpHusky/AndroidManifest.xml
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+    SPDX-FileCopyrightText: The LineageOS Project
+    SPDX-License-Identifier: Apache-2.0
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="org.lineageos.updater.overlay.derp.husky">
+    <overlay
+        android:isStatic="true"
+        android:targetPackage="org.lineageos.updater" />
+</manifest>
diff --git a/rro_overlays/UpdaterOverlayDerpHusky/res/values/strings.xml b/rro_overlays/UpdaterOverlayDerpHusky/res/values/strings.xml
new file mode 100644
index 00000000..f785599a
--- /dev/null
+++ b/rro_overlays/UpdaterOverlayDerpHusky/res/values/strings.xml
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+    SPDX-FileCopyrightText: The LineageOS Project
+    SPDX-License-Identifier: Apache-2.0
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- Unofficial build: DerpFest's own husky.json lists the official builds, which are signed
+         with other keys. Updates come from our OTA repo instead. -->
+    <string name="updater_server_url" translatable="false">https://raw.githubusercontent.com/denson9874/evolution_ota/cnb/derpfest/builds/{device}.json</string>
+    <string name="menu_changelog_url" translatable="false">https://raw.githubusercontent.com/denson9874/evolution_ota/cnb/derpfest/changelogs/<xliff:g id="device_name">%1$s</xliff:g>.txt</string>
+</resources>
-- 
2.43.0

